plugins/hypervisors/vmware/src/main/java/com/cloud/hypervisor/vmware/resource/VmwareResource.java

```diff
@@ -7501,7 +7501,7 @@ private Answer execute(PrepareUnmanageVMInstanceCommand cmd) {
                             VolumeObjectTO newVol = new VolumeObjectTO();
                             newVol.setDataStoreUuid(entry.second().getUuid());
                             String newPath = vmMo.getVmdkFileBaseName(disk);
-                            ManagedObjectReference morDs = HypervisorHostHelper.findDatastoreWithBackwardsCompatibility(targetHyperHost, entry.second().getUuid());
+                            ManagedObjectReference morDs = HypervisorHostHelper.findDatastoreWithBackwardsCompatibility(targetHyperHost != null ? targetHyperHost : sourceHyperHost, entry.second().getUuid());
```

Comment:
@shwstppr might sourceHyperHost be null as well ?

Reply:
@weizhouapache Method gets `sourceHyperHost` either from caller or the host which is processing the command,
https://github.com/apache/cloudstack/pull/5170/files#diff-77635c5f7eb9cef514aaf69e2fdc9217619afc9005bf8db9585c176e7401a88bR7329-R7331
Cannot be null if the command is being processed

Reply:
@shwstppr great, thanks.

no idea if it breaks some features. at least the NPE is fixed by this pr.

Reply:
Above ^^ is my understanding. Let me know if there should be additional checks @weizhouapache

Reply:
Change was originally introduced here, https://github.com/apache/cloudstack/commit/d2ab350a40cffe4eacf625e1648db7c12341f2cb

Reply:
@shwstppr if the origin is a merge commit than this should be fixed on the older branch and merged forward.